test/mockupdb/test_mongos_command_read_mode.py

```diff
@@ -39,18 +39,14 @@ def test_aggregate(self):
         with going(collection.aggregate, []):
             command = server.receives(aggregate='collection', pipeline=[])
             self.assertFalse(command.slave_ok, 'SlaveOkay set')
-            self.assertNotIn('$readPreference', command)
             command.ok(result=[{}])
 
         secondary_collection = collection.with_options(
             read_preference=ReadPreference.SECONDARY)
 
         with going(secondary_collection.aggregate, []):
-            command = server.receives(
-                {'$query': SON([('aggregate', 'collection'),
-                                ('pipeline', []),
-                                ('cursor', {})]),
-                 '$readPreference': {'mode': 'secondary'}})
+
+            command = server.receives(OpMsg({"aggregate": "collection", "pipeline": []}))
```

Comment:
`'$readPreference': {'mode': 'secondary'}})` is an important part of this test, let's add it back.